packages/build/src/core/main.js

```diff
@@ -71,19 +71,23 @@ const build = async function (flags = {}) {
       framework,
       dry,
       siteInfo,
-      telemetry,
       mode,
       logs,
       timers,
       durationNs,
       testOpts,
       statsdOpts,
     })
-    const { success, severityCode } = getSeverity('success')
+    const { success, severityCode, status } = getSeverity('success')
+    await trackBuildComplete({ status, commandsCount, netlifyConfig, durationNs, siteInfo, telemetry, testOpts })
     return { success, severityCode, logs }
   } catch (error) {
-    const { severity } = await handleBuildError(error, errorParams)
-    const { success, severityCode } = getSeverity(severity)
+    const {
+      severity,
+      errorInfo: { netlifyConfig, siteInfo },
+    } = await handleBuildError(error, errorParams)
+    const { success, severityCode, status } = getSeverity(severity)
+    await trackBuildComplete({ status, netlifyConfig, siteInfo, telemetry, testOpts })
```

Comment:
Should we handle errors in `trackBuildComplete()`?
At the moment, this is outside of the main `try`/`catch`, so errors should be ignored. 
Also, the top-level `try`/`catch` currently guarantees that the top-level function never throws.

One possible solution could be to split the `parseErrorInfo()` bit outside of `handleBuildError()`. Like this, `getSeverity()` and `trackBuildComplete()` could be called before `handleBuildError()`. For example:

```js
try {
   ...
} catch (error) {
    const {
      severity,
      errorInfo: { netlifyConfig, siteInfo },
    } = parseErrorInfo(error)
    const { success, severityCode, status } = getSeverity(severity)

    try {
      await trackBuildComplete({ status, netlifyConfig, siteInfo, telemetry, testOpts })
    } catch (error_) {
      await handleBuildError(error_, errorParams)
      return
    }

    await handleBuildError(error, errorParams)
}
```

Reply:
I've moved the error handling to the top level of `trackBuildComplete` to make it more clear that it shouldn't error under normal conditions. We can work on a better error handling approach for telemetry in a later PR.

Reply:
:+1: